### PR TITLE
fix: cherry-pick (#8727)

### PR DESCRIPTION
**Pull request by blackdevelopa**
Cherry-pick fix for https://github.com/MetaMask/metamask-mobile/pull/8727 on version 7.17.0.

### CODE REVIEW

File: app/store/migrations/031.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    },
+  },
+};
+
+jest.mock('@sentry/react-native', () => ({
+  captureException: jest.fn(),
+}));
+const mockedCaptureException = jest.mocked(captureException);
+
+describe('Migration #31', () => {
```
Comment (sethkfman) [inline app/store/migrations/031.test.ts:21]: All values should be updated to `#30`

File: app/store/migrations/031.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    captureException(
+      new Error(`Migration 31: Invalid root state: '${typeof state}'`),
+    );
+    return state;
+  }
+
+  if (!isObject(state.engine)) {
+    captureException(
+      new Error(
+        `Migration 31: Invalid root engine state: '${typeof state.engine}'`,
```
Comment (sethkfman) [inline app/store/migrations/031.ts:21]: All values should be updated to #30

File: app/store/migrations/index.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ import migration26 from './026';
 import migration27 from './027';
 import migration28 from './028';
 import migration29 from './029';
+import migration31 from './031';
```
Comment (sethkfman) [inline app/store/migrations/index.ts:33]: Same as comment above.

File: app/store/migrations/031.test.ts (hunk 1 above)
Comment (blackdevelopa) [reply app/store/migrations/031.test.ts]: Thank you. Updated now to [#30](https://github.com/MetaMask/metamask-mobile/pull/8766/commits/cdf411306ba3e022dd2fd3652e0bc3524ad3bad0). https://github.com/MetaMask/metamask-mobile/pull/8766/commits/cdf411306ba3e022dd2fd3652e0bc3524ad3bad0

**Comment by seaona** [comment]
:green_circle: from QA looks good:
- :heavy_check_mark:  Blockaid is enabled by default on the migration flow

https://github.com/MetaMask/metamask-mobile/assets/54408225/6bacae08-a9a1-4b43-b56b-d9cbaa204604



- :heavy_check_mark:  Blockaid is enabled by default on a fresh install

https://github.com/MetaMask/metamask-mobile/assets/54408225/981003f2-dd5f-4a2f-ac10-b8548a82a48c

**Review by jpuri — APPROVED** [review]

**Pull request merged**